pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -146,8 +144,7 @@ def qfunc(x, y, z):
             # If we did merge, look now at the next gate
             next_gate_idx = find_next_gate(current_gate.wires, list_copy[1:])
 
-        if not allclose(cumulative_angles, zeros(len(cumulative_angles)), atol=atol, rtol=0):
-            current_gate.__class__(*cumulative_angles, wires=current_gate.wires)
+        current_gate.__class__(*cumulative_angles, wires=current_gate.wires)
```

Comment:
Thinking about this again, we could actually add a check here (and in `single_qubit_fusion`) to see if the angles are abstract, and if they are not, do the check for being close to 0 and choose whether to apply the rotation. I'm thinking though that (a) this is super uncommon, (b) for the frequency that it happens, always having to check will not make up for the savings, and (c) we will have to keep a tolerance argument that serves no purpose when jitting, and that might be confusing. What do you think?